checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -107,5 +115,38 @@ def _handle_static_statement(self, block_index: int) -> Optional[list[str] | dic
             return self._handle_static_count_statement(count_statement)
         return
 
-    def _create_new_foreach_resources(self, block_index: int, foreach_statement: list[str] | dict[str, Any]) -> None:
+    def _handle_dynamic_statement(self, blocks_to_render: list[int]) -> dict[int, Optional[list[str] | dict[str, Any] | int]]:
+        rendered_statements_by_idx: dict[int, Optional[list[str] | dict[str, Any] | int]] = {}
```

Comment:
perhaps create an alias to this type and call it something like `FOR_EACH_BLOCK_TYPE`?